httpclient/logging_round_tripper.go

```diff
@@ -0,0 +1,150 @@
+	"net/http"
+	"time"
+)
+
+// LoggingMode represents a mode of logging.
+type LoggingMode string
+
+const (
+	LoggingModeNone   LoggingMode = "none"
+	LoggingModeAll    LoggingMode = "all"
+	LoggingModeFailed LoggingMode = "failed"
+)
+
+// IsValid checks if the logger mode is valid.
+func (lm LoggingMode) IsValid() bool {
+	switch lm {
+	case LoggingModeNone, LoggingModeAll, LoggingModeFailed:
+		return true
+	}
+	return false
+}
+
+// LoggingRoundTripper implements http.RoundTripper for logging requests.
+type LoggingRoundTripper struct {
+	// Delegate is the next RoundTripper in the chain.
+	Delegate http.RoundTripper
+
+	// ReqType is a type of request. e.g. service 'auth-service', an action 'login' or specific information to correlate.
+	ReqType string
+
+	// Mode of logging: none, all, failed.
+	Mode LoggingMode
+
+	// SlowRequestThreshold is a threshold for slow requests.
+	SlowRequestThreshold time.Duration
+
+	// LoggerProvider is a function that provides a context-specific logger.
+	// middleware.GetLoggerFromContext is used by default.
+	LoggerProvider func(ctx context.Context) log.FieldLogger
+}
+
+// LoggingRoundTripperOpts represents an options for LoggingRoundTripper.
+type LoggingRoundTripperOpts struct {
+	// ReqType is a type of request. e.g. service 'auth-service', an action 'login' or specific information to correlate.
+	ReqType string
+
+	// Mode of logging: none, all, failed.
+	Mode LoggingMode
+
+	// SlowRequestThreshold is a threshold for slow requests.
+	SlowRequestThreshold time.Duration
+
+	// LoggerProvider is a function that provides a context-specific logger.
+	// middleware.GetLoggerFromContext is used by default.
+	LoggerProvider func(ctx context.Context) log.FieldLogger
+}
+
+// NewLoggingRoundTripper creates an HTTP transport that log requests.
+func NewLoggingRoundTripper(delegate http.RoundTripper) http.RoundTripper {
+	return NewLoggingRoundTripperWithOpts(delegate, LoggingRoundTripperOpts{
+		ReqType: DefaultReqType,
+	})
+}
+
+// NewLoggingRoundTripperWithOpts creates an HTTP transport that log requests with options.
+func NewLoggingRoundTripperWithOpts(
+	delegate http.RoundTripper, opts LoggingRoundTripperOpts,
+) http.RoundTripper {
+	reqType := opts.ReqType
+	if reqType == "" {
+		reqType = DefaultReqType
+	}
+
+	return &LoggingRoundTripper{
+		Delegate:             delegate,
+		ReqType:              reqType,
+		Mode:                 opts.Mode,
+		SlowRequestThreshold: opts.SlowRequestThreshold,
+		LoggerProvider:       opts.LoggerProvider,
+	}
+}
+
+// getLogger returns a logger from the context or from the options.
+func (rt *LoggingRoundTripper) getLogger(ctx context.Context) log.FieldLogger {
+	if rt.LoggerProvider != nil {
+		return rt.LoggerProvider(ctx)
+	}
+
+	return middleware.GetLoggerFromContext(ctx)
+}
+
+// RoundTrip adds logging capabilities to the HTTP transport.
+func (rt *LoggingRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	if rt.Mode == LoggingModeNone {
```

Comment:
It seems that `LoggingModeNone` is unnecessary since we already have the `log.enabled` configuration parameter. I suggest removing it.